inloop/solutions/views.py

```diff
@@ -126,3 +138,24 @@ def get_context_data(self):
     def get_object(self, **kwargs):
         self.solution = get_object_or_404(Solution, pk=kwargs["id"])
         return self.solution
+
+
+class SolutionFileView(LoginRequiredMixin, View):
+    def get(self, request, **kwargs):
+        task = Task.objects.published().filter(slug=kwargs["slug"])
+        solution = get_object_or_404(
+            Solution, author=self.request.user, task=task, scoped_id=kwargs["scoped_id"]
+        )
+        for solutionfile in solution.solutionfile_set.all():
+            if solutionfile.name == kwargs["title"]:
+                upfile = solutionfile
+                break
+
```

Comment:
What happens if `title` didn't match any of the files in the set? This should result in a 404 page.